sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -117,6 +121,16 @@ private static void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 messageArgs: MemberNameToMessageArguments[memberAccessNameName]));
         }
 
+        private static bool IsResultInContinueWithCall(string memberAccessName, MemberAccessExpressionSyntax memberAccess) =>
+            memberAccessName == ResultName &&
+            memberAccess.Expression is IdentifierNameSyntax identifierNameSyntax &&
+            memberAccess.FirstAncestorOrSelf<InvocationExpressionSyntax>(invocation => IsContinueWithCallWithArgumentName(invocation, identifierNameSyntax.GetName())) != null;
+
+        private static bool IsContinueWithCallWithArgumentName(InvocationExpressionSyntax invocation, string argumentName) =>
+            invocation.Expression.NameIs(ContinueWithName) &&
+            invocation.ArgumentList.Arguments.Any(argument => argument.Expression is SimpleLambdaExpressionSyntax lambda &&
```

Comment:
We need to check `ParenthesizedLambdaExpression` as well. As I recently did this useless parenthesis:
``` C#
return task.ContinueWith((completedTask) => ...
```
Please add test as well.
It would be actually a nice C#-specific code quality rule idea to remove the unneeded parenthesis 😄

Reply:
Good point. I will add a test for that. Also I think it would be a nice rule.